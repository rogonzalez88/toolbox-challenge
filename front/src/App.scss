@import "~bootstrap/scss/bootstrap";

#app {
  nav {
    background-color: #ff6666 !important;
    color: #fff;
    margin-bottom: 10px;
    .navbar-brand {
      color: #fff;
    }
  }
  thead {
    border-bottom: 2px solid #000;
  }
}