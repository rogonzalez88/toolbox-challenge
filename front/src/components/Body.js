import React, { useEffect } from 'react';

import { Table, Spinner } from 'react-bootstrap';
import { useSelector, useDispatch } from 'react-redux';

import { fetchFileData } from '../store/slices/filesSlice';

const Body = () => {
  const dispatch = useDispatch();
  const { tableData, isLoadingTableData } = useSelector(({ files }) => files);
  
  useEffect(() => {
    dispatch(fetchFileData());
  }, [dispatch]);

  if (isLoadingTableData)
    return (
      <div className="d-flex justify-content-center">
        <Spinner animation="border" />
      </div>
    );

  const tableBody = tableData
    .map((file) => file)
    .map(({ lines }) => lines)
    .flat();

  return (
    <Table striped bordered hover>
      <thead>
        <tr>
          {tableData.length > 0 && tableData[0].lines.length > 0 ? (
            Object.keys(tableData[0].lines[0]).map((header, index) => (
              <th className="text-capitalize" key={index}>
                {header}
              </th>
            ))
          ) : (
            <th>No Results</th>
          )}
        </tr>
      </thead>
      <tbody>
        {tableBody.map((content, index) => (
          <tr key={index}>
            {Object.values(content).map((value, valueIndex) => (
              <td key={valueIndex}>{value}</td>
            ))}
          </tr>
        ))}
      </tbody>
    </Table>
  );
};

export default Body;
