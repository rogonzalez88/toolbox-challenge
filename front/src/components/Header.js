import React, { useEffect } from 'react';

import { Container, Navbar, Form } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';

import { fetchFileList, fetchFileData } from '../store/slices/filesSlice';

const Header = () => {
  const dispatch = useDispatch();
  const { fileList, isLoadingFileList } = useSelector((state) => state.files);

  useEffect(() => {
    dispatch(fetchFileList());
  }, [dispatch]);

  const handleFilter = (newValue) => {
    if (newValue === 'all') {
      dispatch(fetchFileData());
    } else {
      dispatch(fetchFileData({ fileName: newValue }));
    }
    
  };

  return (
    <Navbar bg="light" expand="lg" fluid>
      <Container fluid>
        <Navbar.Brand>React Test App</Navbar.Brand>
        <Navbar.Toggle aria-controls="navbarScroll" />
        <Navbar.Collapse id="navbarScroll" className="justify-content-end">
          <Form.Group className="d-flex align-items-center w-25">
            <Form.Label className="w-50">Filter by </Form.Label>
            <Form.Select
              onChange={(event) => handleFilter(event.target.value)}
              disabled={isLoadingFileList}
              defaultValue="select"
            >
              <option value="select" disabled>
                Select file name
              </option>
              {fileList &&
                fileList.files.map((file, index) => (
                  <option value={file} key={index}>
                    {file}
                  </option>
                ))}
              <option value="all">
                All
              </option>
            </Form.Select>
          </Form.Group>
        </Navbar.Collapse>
      </Container>
    </Navbar>
  );
};

export default Header;
