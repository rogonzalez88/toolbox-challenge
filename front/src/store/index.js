import { configureStore } from "@reduxjs/toolkit";

import filesReducer from "./slices/filesSlice";

export default configureStore({
  reducer: {
    files: filesReducer,
  }
});
