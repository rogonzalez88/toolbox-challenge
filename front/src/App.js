import React from 'react';
import { Header, Table } from './components';
import { Container } from 'react-bootstrap';

const App = () => {
  return (
    <Container fluid>
      <Header />
      <Container>
        <Table />
      </Container>
    </Container>
  );
};

export default App;
