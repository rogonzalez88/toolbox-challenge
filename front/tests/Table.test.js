import React from 'react';

import renderer from "react-test-renderer";
import { Table } from "../src/components";
import WrapperReduxTest from "./utils/WrapperReduxTest";

it("Test Table component", () => {
  const component = renderer.create(
    <WrapperReduxTest
      initialState={{
        tableData: [
          {
            name: "test2.csv",
            lines: [
              {
                file: "test2.csv",
                text: "tqKxqXUH",
                number: "680974973",
                hex: "5e9cf3e506b04f0f7f48085451446dce",
              },
            ],
          },
          {
            name: "test3.csv",
            lines: [
              {
                file: "test3.csv",
                text: "HebylCEvV",
                number: "48412389674640590714593150492362",
                hex: "747fe600c914195ba929e13a186bdb8c",
              },
              {
                file: "test3.csv",
                text: "BNMncTytoBhrokGimNYLj",
                number: "55557",
                hex: "c6f9121cb0f4f5b12d75d195775009c3",
              },
              {
                file: "test3.csv",
                text: "cxgcQptaopPzlXhfdfoYsXtjvHt",
                number: "23712",
                hex: "fb70c9c2f77f089d98ea96ac845865c3",
              },
            ],
          },
          {
            name: "test6.csv",
            lines: [
              {
                file: "test6.csv",
                text: "qQgxysMGOUS",
                number: "37354515",
                hex: "82a8617c78cb66e71195bd9e8cb48148",
              },
              {
                file: "test6.csv",
                text: "VXROxXvNHFHbzkvVCOyNRPlcwK",
                number: "998522",
                hex: "911ff44c23f2396fd4b9c51ec6ffdabf",
              },
              {
                file: "test6.csv",
                text: "MvfONHGpFjkhcljtdN",
                number: "29523789124890379908867190832017",
                hex: "b6b806bdf56ac1eefccd4153bd7c19cf",
              },
              {
                file: "test6.csv",
                text: "iYfe",
                number: "942571",
                hex: "ef6a04d2b5c9d753c9d33719fcbaca25",
              },
              {
                file: "test6.csv",
                text: "azQrlVVCJHwxBbNrllShTjgOIHTzP",
                number: "1751438",
                hex: "9447fc4bf98be72b73c0acdf3fd755f1",
              },
              {
                file: "test6.csv",
                text: "RioTvLjPNgrnBogoZJOrMdKWGog",
                number: "3309",
                hex: "6a1948c3645288377be04998012195bc",
              },
              {
                file: "test6.csv",
                text: "rntMJ",
                number: "62985826",
                hex: "ff8ddb5ea003924643ffe991a581dcde",
              },
            ],
          },
        ],
        isLoadingTableData: false,
      }}
    >
      <Table />
    </WrapperReduxTest>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("Test empty Table component", () => {
  const component = renderer.create(
    <WrapperReduxTest
      initialState={{
        isLoadingTableData: false,
      }}
    >
      <Table />
    </WrapperReduxTest>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("Test first render of Table component", () => {
  const component = renderer.create(
    <WrapperReduxTest>
      <Table />
    </WrapperReduxTest>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
