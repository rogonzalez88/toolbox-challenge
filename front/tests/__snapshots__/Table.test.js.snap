// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Table component 1`] = `
<table
  className="table table-striped table-bordered table-hover"
>
  <thead>
    <tr>
      <th
        className="text-capitalize"
      >
        file
      </th>
      <th
        className="text-capitalize"
      >
        text
      </th>
      <th
        className="text-capitalize"
      >
        number
      </th>
      <th
        className="text-capitalize"
      >
        hex
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        test2.csv
      </td>
      <td>
        tqKxqXUH
      </td>
      <td>
        680974973
      </td>
      <td>
        5e9cf3e506b04f0f7f48085451446dce
      </td>
    </tr>
    <tr>
      <td>
        test3.csv
      </td>
      <td>
        HebylCEvV
      </td>
      <td>
        48412389674640590714593150492362
      </td>
      <td>
        747fe600c914195ba929e13a186bdb8c
      </td>
    </tr>
    <tr>
      <td>
        test3.csv
      </td>
      <td>
        BNMncTytoBhrokGimNYLj
      </td>
      <td>
        55557
      </td>
      <td>
        c6f9121cb0f4f5b12d75d195775009c3
      </td>
    </tr>
    <tr>
      <td>
        test3.csv
      </td>
      <td>
        cxgcQptaopPzlXhfdfoYsXtjvHt
      </td>
      <td>
        23712
      </td>
      <td>
        fb70c9c2f77f089d98ea96ac845865c3
      </td>
    </tr>
    <tr>
      <td>
        test6.csv
      </td>
      <td>
        qQgxysMGOUS
      </td>
      <td>
        37354515
      </td>
      <td>
        82a8617c78cb66e71195bd9e8cb48148
      </td>
    </tr>
    <tr>
      <td>
        test6.csv
      </td>
      <td>
        VXROxXvNHFHbzkvVCOyNRPlcwK
      </td>
      <td>
        998522
      </td>
      <td>
        911ff44c23f2396fd4b9c51ec6ffdabf
      </td>
    </tr>
    <tr>
      <td>
        test6.csv
      </td>
      <td>
        MvfONHGpFjkhcljtdN
      </td>
      <td>
        29523789124890379908867190832017
      </td>
      <td>
        b6b806bdf56ac1eefccd4153bd7c19cf
      </td>
    </tr>
    <tr>
      <td>
        test6.csv
      </td>
      <td>
        iYfe
      </td>
      <td>
        942571
      </td>
      <td>
        ef6a04d2b5c9d753c9d33719fcbaca25
      </td>
    </tr>
    <tr>
      <td>
        test6.csv
      </td>
      <td>
        azQrlVVCJHwxBbNrllShTjgOIHTzP
      </td>
      <td>
        1751438
      </td>
      <td>
        9447fc4bf98be72b73c0acdf3fd755f1
      </td>
    </tr>
    <tr>
      <td>
        test6.csv
      </td>
      <td>
        RioTvLjPNgrnBogoZJOrMdKWGog
      </td>
      <td>
        3309
      </td>
      <td>
        6a1948c3645288377be04998012195bc
      </td>
    </tr>
    <tr>
      <td>
        test6.csv
      </td>
      <td>
        rntMJ
      </td>
      <td>
        62985826
      </td>
      <td>
        ff8ddb5ea003924643ffe991a581dcde
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Test empty Table component 1`] = `
<table
  className="table table-striped table-bordered table-hover"
>
  <thead>
    <tr>
      <th>
        No Results
      </th>
    </tr>
  </thead>
  <tbody />
</table>
`;

exports[`Test first render of Table component 1`] = `
<div
  className="d-flex justify-content-center"
>
  <div
    className="spinner-border"
  />
</div>
`;
